kie-api/src/main/java/org/kie/api/internal/utils/ServiceDiscoveryImpl.java

```diff
@@ -237,4 +252,58 @@ public void put(int priority, K key, V value) {
             return map.entrySet();
         }
     }
-}
+
+    public static Enumeration<URL> findKieConfUrls(ClassLoader cl) throws IOException {
+        List<URL> kieConfsUrls = new ArrayList<>();
+
+        Enumeration<URL> metaInfs = cl.getResources(CONF_FILE_FOLDER);
+        while (metaInfs.hasMoreElements()) {
+            URL metaInf = metaInfs.nextElement();
+            if (metaInf.getProtocol().startsWith("vfs")) {
+                kieConfsUrls = Stream.of(KIE_MODULES)
```

Comment:
Warn the user in case the system is working outside OSGi or JBoss but will fail inside it.